modules/api/pkg/ee/resource-quota/handler.go

```diff
@@ -172,35 +214,373 @@ func GetResourceQuota(ctx context.Context, request interface{}, provider provide
+		calculatedResources.CPU.Cmp(*projectResourceQuota.Spec.Quota.CPU) > 0 {
+		msg += fmt.Sprintf("Calculated cpu (%s) exceeds resource quota (%s)", calculatedResources.CPU, projectResourceQuota.Spec.Quota.CPU)
+	}
+	if projectResourceQuota.Spec.Quota.Memory != nil && calculatedResources.Memory != nil &&
+		calculatedResources.Memory.Cmp(*projectResourceQuota.Spec.Quota.Memory) > 0 {
+		msg += fmt.Sprintf("Calculated memory (%s) exceeds resource quota (%s)", calculatedResources.Memory, projectResourceQuota.Spec.Quota.Memory)
+	}
+	if projectResourceQuota.Spec.Quota.Storage != nil && calculatedResources.Storage != nil &&
+		calculatedResources.Storage.Cmp(*projectResourceQuota.Spec.Quota.Storage) > 0 {
+		msg += fmt.Sprintf("Calculated disk size (%s) exceeds resource quota (%s)", calculatedResources.Storage, projectResourceQuota.Spec.Quota.Storage)
+	}
+
+	return &apiv2.ResourceQuotaUpdateCalculation{
+		ResourceQuota:   *convertToAPIStruct(projectResourceQuota, projectName),
+		CalculatedQuota: convertToAPIQuota(*calculatedResources),
+		Message:         msg,
+	}, nil
+}
+
+func getResourceDetailsFromRequest(req calculateProjectResourceQuotaUpdate) (*kubermaticv1.ResourceDetails, error) {
+	nc := &kubermaticprovider.NodeCapacity{}
+
+	var err error
+
+	switch {
+	case req.Body.AlibabaInstanceType != nil:
+		if err = getAlibabaResourceDetails(req, nc); err != nil {
+			return nil, err
+		}
+	case req.Body.AnexiaNodeSpec != nil:
+		if err = getAnexiaResourceDetails(req, nc); err != nil {
+			return nil, err
+		}
+	case req.Body.AWSSize != nil:
+		if err = getAWSResourceDetails(req, nc); err != nil {
+			return nil, err
+		}
+	case req.Body.AzureSize != nil:
+		if err = getAzureResourceDetails(req, nc); err != nil {
+			return nil, err
+		}
+	case req.Body.DOSize != nil:
+		if err = getDOResourceDetails(req, nc); err != nil {
+			return nil, err
+		}
+	case req.Body.EquinixSize != nil:
+		if err = getEquinixResourceDetails(req, nc); err != nil {
+			return nil, err
+		}
+	case req.Body.GCPSize != nil:
+		if err = getGCPResourceDetails(req, nc); err != nil {
+			return nil, err
+		}
+	case req.Body.HetznerSize != nil:
+		if err = getHetznerResourceDetails(req, nc); err != nil {
+			return nil, err
+		}
+	case req.Body.NutanixNodeSpec != nil:
+		if err = getNutanixResourceDetails(req, nc); err != nil {
+			return nil, err
+		}
+	case req.Body.OpenstackSize != nil:
+		if err = getOpenstackResourceDetails(req, nc); err != nil {
+			return nil, err
+		}
+	case req.Body.VSphereNodeSpec != nil:
+		if err = getVSphereResourceDetails(req, nc); err != nil {
+			return nil, err
+		}
+	case req.Body.VMDirectorNodeSpec != nil:
+		if err = getVMCloudDirectorResourceDetails(req, nc); err != nil {
+			return nil, err
+		}
+	default:
+		return nil, fmt.Errorf("provider set in request not supported: %v", req.Body)
+	}
+
+	// These should not happen, unless we have a bug in the setting resource values code above
```

Comment:
Is it because provider's cpu count is always an `int` and it's defaulted to 0?

Reply:
Changed it now to default the values. It seems more intuitive then throwing an error, as the implementations for providers can vary